src/WinRT.Runtime/TypeExtensions.cs

```diff
@@ -155,23 +149,49 @@ public static Type GetGuidType(this Type type)
             }
 #endif
 
-            Type vftblType = helperType.GetNestedType("Vftbl");
-            if (vftblType is null)
-            {
-                return null;
-            }
-            if (helperType.IsGenericType && vftblType is object)
+#if NET8_0_OR_GREATER
+            [RequiresDynamicCode(AttributeMessages.MarshallingOrGenericInstantiationsRequiresDynamicCode)]
+#endif
+#if NET
+            [DynamicDependency(DynamicallyAccessedMemberTypes.All, "ABI.Windows.Foundation.IAsyncActionWithProgress`1+Vftbl", "Microsoft.Windows.SDK.NET")]
+            [DynamicDependency(DynamicallyAccessedMemberTypes.All, "ABI.Windows.Foundation.IAsyncOperationWithProgress`2+Vftbl", "Microsoft.Windows.SDK.NET")]
+            [DynamicDependency(DynamicallyAccessedMemberTypes.All, "ABI.Windows.Foundation.IAsyncOperation`1+Vftbl", "Microsoft.Windows.SDK.NET")]
+            [DynamicDependency(DynamicallyAccessedMemberTypes.All, "ABI.Windows.Foundation.Collections.IMapChangedEventArgs`1+Vftbl", "Microsoft.Windows.SDK.NET")]
+            [DynamicDependency(DynamicallyAccessedMemberTypes.All, "ABI.Windows.Foundation.Collections.IObservableMap`2+Vftbl", "Microsoft.Windows.SDK.NET")]
+            [DynamicDependency(DynamicallyAccessedMemberTypes.All, "ABI.Windows.Foundation.Collections.IObservableVector`1+Vftbl", "Microsoft.Windows.SDK.NET")]
+            [DynamicDependency(DynamicallyAccessedMemberTypes.All, "ABI.System.EventHandler`1+Vftbl", "WinRT.Runtime")]
+            [DynamicDependency(DynamicallyAccessedMemberTypes.All, "ABI.System.System.Collections.Generic.KeyValuePair`2+Vftbl", "WinRT.Runtime")]
+            [DynamicDependency(DynamicallyAccessedMemberTypes.All, "ABI.System.System.Collections.Generic.IEnumerable`1+Vftbl", "WinRT.Runtime")]
+            [DynamicDependency(DynamicallyAccessedMemberTypes.All, "ABI.System.System.Collections.Generic.IEnumerator`1+Vftbl", "WinRT.Runtime")]
+            [DynamicDependency(DynamicallyAccessedMemberTypes.All, "ABI.System.System.Collections.Generic.IList`1+Vftbl", "WinRT.Runtime")]
+            [DynamicDependency(DynamicallyAccessedMemberTypes.All, "ABI.System.System.Collections.Generic.IReadOnlyList`1+Vftbl", "WinRT.Runtime")]
+            [DynamicDependency(DynamicallyAccessedMemberTypes.All, "ABI.System.System.Collections.Generic.IDictionary`2+Vftbl", "WinRT.Runtime")]
+            [DynamicDependency(DynamicallyAccessedMemberTypes.All, "ABI.System.System.Collections.Generic.IReadOnlyDictionary`2+Vftbl", "WinRT.Runtime")]

```

Comment:
Is this list fixed or can it grow with additional custom projections?

For the types defined outside of WinRT.Runtime, is this temporary until an updated projection is produced?

What about embedded-CsWinRT scenarios? Those would need different assembly names here.

Reply:
This list is fixed, it is only for generic interfaces and developers can't define their own generic interface.  This is just for compat with pre-existing projections being used in apps that have trimming enabled and have an updated Windows SDK projection / WinRT.Runtime.  All of the newly generated projections will not need the vtbl.  This also means for embedded scenarios, if they have this it means the updated version of CsWinRT is being used to generate the embedded projection and thereby it should be using the newly generated projection which doesn't need the vftbl type.